javascript-sdk/src/sdk.ts

```diff
@@ -169,48 +162,27 @@ export default class RebuffSdk implements Rebuff {
     return false;
   }
 
-  // Calling functions immediately after constructor can cause issues if pinecone connection needs time
-  async getPineconeConnection(): Promise<PineconeClient> {
-    if (this.pinecone.conn) {
-      return this.pinecone.conn;
+  // Calling functions immediately after constructor can cause issues if vector store needs time
+  async getVectorStore(): Promise<VectorStore> {
+    if (this.vectorStore) {
+      return this.vectorStore;
     }
-    // Wait 1 second for pinecone connection to be initialized by constructor
+    // Wait 1 second for vector store to be initialized by constructor
     await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
Setting a timeout seems like it could be flaky. The vectorStore type is also incorrect since it can be undefined. Instead of checking for initialization, waiting, and failing again if it's not initialized, I think we could fix this with the following pattern:

```
  private vectorStore: VectorStore | undefined;
  private sdkConfig: SdkConfig;
```

Constructor:
```
  constructor(config: SdkConfig) {
    this.sdkConfig = config;
    this.openai = {
      conn: getOpenAIInstance(config.openai.apikey),
      model: config.openai.model || "gpt-3.5-turbo",
    };
  }
```

getVectorStore: 
```
  async getVectorStore(): Promise<VectorStore> {
    if (this.vectorStore) {
      return this.vectorStore;
    }
    return await initPinecone(
      this.sdkConfig.pinecone.environment,
      this.sdkConfig.pinecone.apikey,
      this.sdkConfig.pinecone.index,
      this.sdkConfig.openai.apikey
    );
  }
```

Reply:
Yeah, I had the same concern about the vector store initialization, but I didn't want to make _too_ many changes on my first PR. I like your suggestion though. :+1: